specutils/io/asdf/tags/tests/test_spectra.py

```diff
@@ -26,27 +27,26 @@ def create_spectrum1d(xmin, xmax, uncertainty=None):
 def test_asdf_spectrum1d(tmpdir):
 
     spectrum = create_spectrum1d(5100, 5300)
+    Spectrum1DType.assert_equal(spectrum, roundtrip_object(spectrum))
 
-    tree = dict(spectrum=spectrum)
-    assert_roundtrip_tree(tree, tmpdir)
 
 
 @pytest.mark.filterwarnings('ignore:ASDF functionality for astropy is being moved out')
 def test_asdf_spectrum1d_uncertainty(tmpdir):
 
     spectrum = create_spectrum1d(5100, 5300, uncertainty=True)
+    Spectrum1DType.assert_equal(spectrum, roundtrip_object(spectrum))
 
-    tree = dict(spectrum=spectrum)
-    assert_roundtrip_tree(tree, tmpdir)
 
 
 @pytest.mark.xfail
 def test_asdf_spectralaxis(tmpdir):
 
     wavelengths  = np.arange(5100, 5300) * 0.1 * u.nm
     spectral_axis = SpectralAxis(wavelengths, bin_specification="edges")
-    tree = dict(spectral_axis=spectral_axis)
-    assert_roundtrip_tree(tree, tmpdir)
+    # there is no implemented asdf type for SpectralAxis and no defined
+    # equality comparison (assert_equal)
+    assert roundtrip_object(spectral_axis) == spectrum_axis
```

Comment:
Will this actually pass now? This test is marked as xfail.

Reply:
Nope, we have xfail strict on and CI passed, so this is still xfail

https://github.com/astropy/specutils/blob/ca3153f05b8d3613bba0b865e273187e8ff176c4/setup.cfg#L52